backend/src/api/article/middlewares/isAuth.js

```
@@ -0,0 +1,136 @@
+      console.log("contentSubscription.stripeProductId", contentSubscription.stripeProductId)
+      console.log("ctx.state.user?.subscription?.plan?.product", ctx.state.user?.subscription?.plan?.product)
+
+      const subscription = await strapi.query('api::subscription.subscription').findOne({
+        where: {
+          stripeProductId: ctx.state.user?.subscription?.plan?.product, // Assuming the subscription has a relation to the user
+        },
+      });
+
+      if (hasMatchingSubscription || (subscription ? subscription.hierarchy === 1 : false)) {
+        removeAdsForSubscribedUser(ctx, ctx.response.body.Content);
+        delete ctx.response.body.subscriptions;  // Remove subscription details from response
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)